kernels/streamer_alu/main.c

```diff
@@ -1,116 +1,65 @@
 #include "snrt.h"
-
+#include "memref.h"
 #include "data.h"
+#include "snax_rt.h"
+
+void _mlir_ciface_streamer_add(OneDMemrefI32_t* A, OneDMemrefI32_t* B, OneDMemrefI32_t* O);
 
 int main() {
   // Set err value for checking
   int err = 0;
 
-  // Allocates space in TCDM
-  uint64_t *local_a, *local_b, *local_o;
-
-  local_a = (uint64_t *)snrt_l1_next();
-  local_b = local_a + DATA_LEN;
-  local_o = local_b + DATA_LEN;
-
-  // Start of pre-loading data from L2 memory
-  // towards the L1 TCDM memory
-  // Use the Snitch core with a DMA
-  // to move the data from L2 to L1
-  if (snrt_is_dm_core()) {
-    // This measures the start of cycle count
-    // for preloading the data to the L1 memory
-    uint32_t start_dma_load = snrt_mcycle();
-
-    // The DATA_LEN is found in data.h
-    size_t vector_size = DATA_LEN * sizeof(uint64_t);
-    snrt_dma_start_1d(local_a, A, vector_size);
-    snrt_dma_start_1d(local_b, B, vector_size);
-
-    // Measure the end of the transfer process
-    uint32_t end_dma_load = snrt_mcycle();
-  }
+  // Create memref objects for data stored in L3
+  OneDMemrefI32_t memrefA;
```

Comment:
using I32 here for I64 and I128, as it does not really matter, I think we can change this to OneDMemref_t maybe to simplify things?

Reply:
In a separate pr, as this affects all of the kernels

Reply:
Sounds like a good idea. We could do this with void pointers. Essentially telling the compiler that we are in _expert mode_ :^)